src/convolution/convolution_2d.jl

```diff
@@ -0,0 +1,1191 @@
+macro bind(def, element)
+    quote
+        local iv = try Base.loaded_modules[Base.PkgId(Base.UUID("6e696c72-6542-2067-7265-42206c756150"), "AbstractPlutoDingetjes")].Bonds.initial_value catch; b -> missing; end
+        local el = $(esc(element))
+        global $(esc(def)) = Core.applicable(Base.get, el) ? Base.get(el) : iv(el)
+        el
+    end
+end
+
+# ╔═╡ 746ff659-88ab-4ff4-8cba-43c798cacd3e
+using Images,TestImages, PlutoUI
+
+# ╔═╡ 04a5c6d4-f8d5-11ed-141a-35481b811ee9
+md"""
+## Images as Lists of numbers
+Hi There! I am Boshra (github: @ariguiba) and I am here to teach you about Images and Filters! 
+
+Remember when you were kid and you used to play with legos? you probably would put tiny pieces of different colors to form different shapes and elements? 
+
+Well images in computers work exactly the same way! Each image is made of tiny elements we call pixels. Since our computers only understand numbers, these pixels are given to the computer as a list of numbers.  
+
+In the following, we will explore how images can be processed in computer science and we will introduce a cool tool called filtering, which is a mathematical operation to modify images by smoothing, highlighting some parts of the images and way more.
+"""
+
+# ╔═╡ 9be00bec-59a5-478b-ada2-854f7a52d66e
+md"""To keep things simple, we will only deal with black and white images for now. So, for a black and white image, each pixel is a number between 0 and 1. 
+
+Try it out! Move the slider around to set the pixel below to get a different shade of gray"""
+
+# ╔═╡ c4da04ab-f7e0-46fd-b352-e518e4733608
+g_slider = @bind g PlutoUI.Slider(0:0.05:1)
+
+# ╔═╡ 96a4b35a-5a3a-4ad0-9ffe-306db46d1c03
+pixel = Gray.(g)
+
+# ╔═╡ 3f830a9b-3a78-4d18-ae3e-933778402a25
+md"""Now we can generate a 5x5 table of random pixels and combine them together and we have already our first image!
+"""
+
+# ╔═╡ 98919284-8099-43c8-b123-d1476d27a88d
+img = rand(Gray, 5, 5)
+
+# ╔═╡ eb983abe-5899-431b-84a1-7d06715d75c4
+md"""
+Now that you know how images are built. Let's get to the fun part. Below is a dropdown with popular test images. Choose the one you like and let's start learning about filters.
+"""
+
+# ╔═╡ 4edbfca5-cb25-4093-9f4d-5cfa2aa8d77f
+@bind test_img_selected Select([
+	testimage("coffee") => "Coffee", 
+	testimage("house") => "House",
+	testimage("pirate") => "Pirate",
+	testimage("peppers") => "Peppers",
+	testimage("mandrill") => "Mandrill"
+	])
+
+# ╔═╡ ef0bea69-6dfd-4526-a96e-6607771660fe
+test_img = Gray.(test_img_selected)
+
+# ╔═╡ ebb63415-2738-47f3-b0ef-0fd6e6bac259
+md"""
+## Using Filter 
+
+Now to make things more interesting, we will add some code that applies a filter to this image.
+
+This code basically goes over all the pixels in the image and applies the mathematical operation of convolution, that is the values of the pixel and its surrounding pixels are multipled by the value of the filter matrix. 
```

Comment:
I think it would be nice if the idea of filters was explained a bit more. Your notebook starts of with a good, accessible explanation of what pixels are, so we can assume that the reader would also never have heard of filters. (It's also confusing because they might be thinking of "instagram filters".)

Here is my suggestion for this line:

```suggestion
A **filter** is a small matrix that describes how we can calculate the new value of the pixel. Our code will go over all the pixels in the image and apply the mathematical operation of **convolution**: the values of the pixel and its surrounding pixels are multiplied by the values in the matrix.
```